src/components/contents/AppWriting/AppWriting.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { VFC } from 'react'
+import styled from 'styled-components'
+import { graphql, useStaticQuery } from 'gatsby'
+import { Text } from 'smarthr-ui'
+import { FragmentTitle } from '../../article/FragmentTitle/FragmentTitle'
+import reactStringReplace from 'react-string-replace'
+import { marked } from 'marked'
+
+const query = graphql`
+  query AppWriting {
+    appWritingData: allAirtable(filter: { table: { eq: "UIテキスト" } }) {
+      edges {
+        node {
+          data {
+            name
+            description
+            discussion
+            source
+            record_id
+            order
+          }
+        }
+      }
+    }
+  }
+`
+export const AppWriting: VFC = () => {
+  const data = useStaticQuery<GatsbyTypes.AppWritingTableQuery>(query)
+
+  const appWritingData = data.appWritingData.edges
+    .map(({ node }) => ({
+      name: node.data?.name,
+      description: marked.parse(node.data?.description || ''),
+      discussion: node.data?.discussion,
+      source: node.data?.source,
+      recordId: node.data?.record_id,
+    }))
+    .reverse()
+
+  const getReplaceLinkText = (text: string) => {
+    return (
+      <Text as="p" whiteSpace="pre-wrap">
+        {reactStringReplace(text, /(https?:\/\/\S+)/g, (match: string, strIndex: number) => {
+          return (
+            <a key={strIndex} href={match}>
+              {match}
+            </a>
+          )
+        })}
+      </Text>
+    )
+  }
+  return (
+    <>
+      {appWritingData
+        .filter((item) => {
+          return item.name
+        })
+        .map(({ name, description, discussion, source, recordId }, index) => {
+          const fragmentId = (suffixId: string) => {
+            return recordId ? `${recordId}-${suffixId}` : ''
+          }
```

Comment:
recordId がない場合に、どの id にも空文字が入るのは問題がありそうな気がします。
recordId が false になる可能性がありますか?

```suggestion
          const fragmentId = (suffixId: string) => {
            return recordId ? `${recordId}-${suffixId}` : undefined
          }
```

あるいは

```ts
if (!recordId) {
  return
}

return `${recordId}-${suffixId}`
```

でもいいかも。